Remotion/Web/Core/UI/SmartPage.cs

```diff
@@ -619,13 +624,23 @@ bool ISmartNavigablePage.IsSmartFocusingEnabled
   /// Gets or sets the flag that determines whether a Content-Security-Policy header is set.
   /// The value of the header is determined by <see cref="GetCspHeader"/>.
   /// </summary>
+  /// <remarks>
+  /// This property cannot be set after <see cref="CreateHtmlTextWriter"/> was called at least once.
+  /// The last chance to set this property is usually in OnPreRender.
+  /// </remarks>
   [Description("The flag that determines whether a Content-Security-Policy header is set.")]
   [Category("Behavior")]
   [DefaultValue(null)]
   public bool? EnableCsp
   {
     get { return _enableCsp; }
-    set { _enableCsp = value; }
+    set
+    {
+      if (_htmlWriterCreated)
+        throw new InvalidOperationException($"{nameof(EnableCsp)} cannot be set after an HtmlTextWriter was already created.");
+
+      _enableCsp = value;
+    }
   }
 
   /// <summary> Gets the evaluated value for the <see cref="EnableCsp"/> property. </summary>
```

Comment:
I thought, but might misremember that we wanted to also only evaluate the default once to prevent the default from changing. please memory-assist